Source/WebCore/platform/graphics/filters/FilterResults.cpp

```diff
@@ -40,8 +42,31 @@ FilterImage* FilterResults::effectResult(FilterEffect& effect) const
     return m_results.get(effect);
 }
 
+size_t FilterResults::memoryCost() const
+{
+    CheckedSize memoryCost;
+
+    for (auto& result : m_results.values()) {
+        memoryCost += result->memoryCost();
+        RELEASE_ASSERT(!memoryCost.hasOverflowed());
+    }
+
+    return memoryCost;
+}
+
+bool FilterResults::canCacheResult(FilterImage& result) const
+{
+    CheckedSize totalMemoryCost = memoryCost() + result.memoryCost();
```

Comment:
The addition could overflow before it is assigned to the `Checked`. Instead maybe:

```
CheckedSize totalMemoryCost;
totalMemoryCost += memoryCost();
totalMemoryCost += result.memoryCost();
```

Reply:
Fixed.